Fix VREffect support for webvr-polyfill

brianchirls opened the pull request:
- fixes #8712
- correctly set `isPresenting` in full screen mode
- webvr-polyfill fires fullscreenchange in VR mode
- fix needed for setting main window to full screen while presenting

brianchirls committed:
Fix VREffect support for webvr-polyfill (#8712)
- correctly set `isPresenting` in full screen mode
- webvr-polyfill fires fullscreenchange in VR mode
- fix needed for setting main window to full screen while presenting

mrdoob commented on examples/js/effects/VREffect.js:
```diff
@@ -102,16 +102,26 @@ THREE.VREffect = function ( renderer, onError ) {
 	document.addEventListener( fullscreenchange, function () {
 
-		isPresenting = isDeprecatedAPI && vrHMD && ( document.mozFullScreenElement || document.webkitFullscreenElement ) !== undefined;
+		isPresenting = vrHMD && ( vrHMD.isPresenting || isDeprecatedAPI && ( document.mozFullScreenElement || document.webkitFullscreenElement ) !== undefined );
```
We should add some parenthesis to make this code more clear...

Are we doing `X && ( ( Y || Z ) && W )` or `X && ( Y || ( Z && W ) )`?

brianchirls replied:
I agree that it's difficult to read. Maybe an `if` statement or two would be clearer. Would you prefer that?

mrdoob replied:
I think adding clarifying parenthesis should be enough...

brianchirls committed:
Code clean-up (#8748, #8712)
- Added parentheses to clarify per @mrdoob
- More spaces
- cast `isPresenting` to boolean

brianchirls commented:
@mrdoob Updated with parentheses and some other clean-up

brianchirls committed:
Check if vrHMD is undefined (#8748, #8712)

brianchirls commented:
Fair enough. Fixed.

mrdoob commented:
Thanks!
